### PR TITLE
Update Fluent-ui/icons package

### PR DESCRIPTION
**Pull request by dmceachernmsft**
# What

Update the `@fluent-ui/icons` package
# Why

Allows us to use the newer icons that are needed for Call-readiness

# How Tested

Visually inspected icons in CallWithChat sample app. CI will validate icons are no different.

### CODE REVIEW

File: packages/communication-react/review/beta/communication-react.api.md [hunk 1]
```diff
@@ -1678,6 +1679,7 @@ export const DEFAULT_COMPOSITE_ICONS: {
     ResumeCall: JSX.Element;
     BackSpace: JSX.Element;
     iconBackdrop: JSX.Element;
+    sparkle: JSX.Element;
```
Comment (edwardlee-msft) [inline packages/communication-react/review/beta/communication-react.api.md:1682]: ✨

Comment (edwardlee-msft) [reply packages/communication-react/review/beta/communication-react.api.md:1682]: hmm I'm interested, what does this sparkle icon look like? 
Can't seem to find it in fluentui icons
image

File: packages/react-components/src/components/DomainPermissions.tsx [hunk 2]
```diff
@@ -58,7 +59,9 @@ const _DomainPermissionsContainer = (props: DomainPermissionsProps): JSX.Element
           <Icon styles={iconPrimaryStyles} iconName={'ControlButtonCameraOn'}></Icon>
           <Icon styles={iconBackDropStyles} iconName={'iconBackdrop'}></Icon>
         </Stack>
-        <Icon styles={iconPrimaryStyles} iconName={'Sparkle'}></Icon>
```
Comment (JamesBurnside) [inline packages/react-components/src/components/DomainPermissions.tsx:61]: This should be its own PR - "updating to use sparkle icon" - this PR should _just_ be the new icon package

File: packages/react-components/src/components/DomainPermissions.tsx [hunk 3]
```diff
@@ -58,7 +59,9 @@ const _DomainPermissionsContainer = (props: DomainPermissionsProps): JSX.Element
...
+        <Stack styles={iconContainerStyles} horizontalAlign={'center'}>
```
Comment (JamesBurnside) [inline packages/react-components/src/components/DomainPermissions.tsx:62]: Do not apply transform here, see comment here: https://teams.microsoft.com/l/message/19:5e150d250f724d1089c949305efdc8cc@thread.skype/1662478489025?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=3e9c1fc3-39df-4486-a26a-456d80e80f82&parentMessageId=1661978742851&teamName=Azure%20Communication%20Services&channelName=Web%20UI%20-%20Team&createdTime=1662478489025&allowXTenantAccess=false

File: packages/communication-react/review/beta/communication-react.api.md (hunk 1 above)
Comment (JamesBurnside) [reply packages/communication-react/review/beta/communication-react.api.md:1682]: Use this site for searching for icons: https://iconcloud.design/browse/Fluent%20System%20Library/Fluent%20Regular

File: packages/react-components/src/components/styles/DomainPermissions.styles.ts [hunk 4]
```diff
@@ -47,6 +47,16 @@ export const iconBackDropStyles: IIconStyles = {
   }
 };
 
+/**
+ * @internal
+ */
+export const sparkleIconBackdropStyles: IIconStyles = {
+  root: {
+    color: '#EFF6FC',
```
Comment (JamesBurnside) [inline packages/react-components/src/components/styles/DomainPermissions.styles.ts:55]: is this a fixed color or a theme color?

Comment (dmceachernmsft) [reply packages/react-components/src/components/styles/DomainPermissions.styles.ts:55]: Fixed colour based on Juan's designs, though he has not given us a dark mode version of what to expect here. He knows about it and will hopefully give us more examples of what this might look like. After today's discussion as well it would seem like we might want to define style types for these components so that we can provide more customization?

File: packages/communication-react/review/beta/communication-react.api.md (hunk 1 above)
Comment (edwardlee-msft) [reply packages/communication-react/review/beta/communication-react.api.md:1682]: Ah ha! Thanks James!
image

File: packages/react-components/src/components/styles/DomainPermissions.styles.ts (hunk 4 above)
Comment (prprabhu-ms) [reply packages/react-components/src/components/styles/DomainPermissions.styles.ts:55]: We pretty much never want fixed colors in our components.

The only exceptions are when we overlay on top of video tile. There, we don't control the background, so we fixed the palette to optimize for a11y.

File: packages/react-components/src/theming/icons.tsx [hunk 5]
```diff
@@ -147,5 +147,7 @@ export const DEFAULT_COMPONENT_ICONS = {
   /* @conditional-compile-remove(dialpad) */ /* @conditional-compile-remove(PSTN-calls) */
   BackSpace: <Backspace20Regular />,
   /* @conditional-compile-remove(call-readiness) */
-  iconBackdrop: <Circle24Filled />
+  iconBackdrop: <Circle24Filled />,
+  /* @conditional-compile-remove(call-readiness) */
+  sparkle: <Sparkle20Filled />
```
Comment (prprabhu-ms) [inline packages/react-components/src/theming/icons.tsx:152]: When you do add this (in a separate PR), it should be named to signify where it isused, not what icon it is.

The names we define should be domain specific. This way Contoso knows what overriding that icon does, and it also avoids conflicts with the default icon names in Fluent.

**Review by prprabhu-ms — COMMENTED** [review]
Start call icon is now better hz aligned. Yay!

image

File: packages/react-components/src/components/styles/DomainPermissions.styles.ts (hunk 4 above)
Comment (dmceachernmsft) [reply packages/react-components/src/components/styles/DomainPermissions.styles.ts:55]: okay I will chat with Juan to see what he is expecting for colours here. I'm not sure if this blue is in the theme but will check

Comment (dmceachernmsft) [reply packages/react-components/src/components/styles/DomainPermissions.styles.ts]: this blue is not in the theme that is provided unfortunately

File: packages/react-components/src/components/DomainPermissions.tsx (hunk 3 above)
Comment (dmceachernmsft) [reply packages/react-components/src/components/DomainPermissions.tsx:61]: Will be done in follup PR: https://github.com/Azure/communication-ui-library/pull/2307

File: packages/react-components/src/components/DomainPermissions.tsx (hunk 2 above)
Comment (dmceachernmsft) [reply packages/react-components/src/components/DomainPermissions.tsx:61]: this PR: https://github.com/Azure/communication-ui-library/pull/2307 😊

File: packages/communication-react/review/beta/communication-react.api.md (hunk 1 above)
Comment (dmceachernmsft) [reply packages/communication-react/review/beta/communication-react.api.md]: being added in followup PR: https://github.com/Azure/communication-ui-library/pull/2307 😊

**Review by JamesBurnside — APPROVED** [review]

**Review by edwardlee-msft — APPROVED** [review]

**Comment by edwardlee-msft** [comment]
Wow, for the life of me I can't tell if anything change between those diffs.
And sadly the alignment is not auto fixed anymore with this specific fluentui update :(.
image

**Review by edwardlee-msft — APPROVED** [review]

**Comment by dmceachernmsft** [comment]
> Wow, for the life of me I can't tell if anything change between those diffs. And sadly the alignment is not auto fixed anymore with this specific fluentui update :(. image

Yeah some of them seem to be logging changes but there isn't anything different. I wonder if the snapshots might change when the DOM structure changes too? I'm sad that we lose the alignment stuff but we do at least get the new icon we were looking for here.

**Review by JamesBurnside — APPROVED** [review]
